lookout/style/typos/corrector.py

```diff
@@ -77,118 +82,110 @@ def initialize_generator(self, vocabulary_file: str, frequencies_file: str,
         Construct a new CandidatesGenerator.
 
         :param vocabulary_file: The path to the vocabulary.
-        :param frequencies_file: The path ot the frequencies.
+        :param frequencies_file: The path to the frequencies.
         :param embeddings_file: The path to the embeddings.
         :param neighbors_number: Number of neighbors of context and typo embeddings \
                                  to consider as candidates.
         :param edit_candidates: Number of the most frequent tokens among tokens on \
                                 equal edit distance from the typo to consider as candidates.
         :param max_distance: Maximum edit distance for symspell lookup.
         :param radius: Maximum edit distance from typo allowed for candidates.
-        :return: Nothing
         """
         self.generator.construct(vocabulary_file, frequencies_file, embeddings_file,
                                  neighbors_number, edit_candidates, max_distance, radius)
 
-    def train(self, typos: pandas.DataFrame, candidates: pandas.DataFrame = None,
+    def train(self, data: pandas.DataFrame, candidates:  str = None,
               save_candidates_file: str = None) -> None:
         """
-        Train corrector on the given dataset of typos inside identifiers.
+        Train corrector on tokens from the given dataset.
 
-        :param typos: DataFrame containing columns "typo" and "identifier",
-                      column "token_split" is optional, but used when present.
-        :param candidates: DataFrame with precalculated candidates.
-        :param save_candidates_file: Path to file where to save the candidates.
+        :param data: DataFrame, containing columns Columns.Token and Columns.CorrectToken, \
+                     column Columns.Split is optional, but used when present.
+        :param candidates: A .csv.xz dump of a dataframe with precalculated candidates.
+        :param save_candidates_file: Path to file where to save the candidates (.csv.xz).
         """
-        if candidates is None:
+        if not candidates:
```

Comment:
The default value is `None` so we should check `None`. An empty string may signal about a problem up the stack so we should consider it as a legitimate value.